app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -0,0 +1,144 @@
+module Mixins
+  module BreadcrumbsMixin
+    def self.included(c)
+      c.helper_method(:data_for_breadcrumbs)
+    end
+
+    # Main method which creates all breadcrumbs and returns them (to view page which will parse them into breadcrumbs nav)
+    def data_for_breadcrumbs
+      options = breadcrumbs_options || {}
+      options[:record_info] ||= (@record || {})
+      # diplay name, description ?
+      options[:record_title] ||= :name
+      breadcrumbs = options[:breadcrumbs] || []
+
+      # Different methods for controller with explorers and for non-explorers controllers
+
+      if @sb[:explorer].blank? && @right_cell_text.blank?
+        # Append breadcrumb from @record item (eg "Openstack")
+        breadcrumbs.push(build_breadcrumbs_no_explorer(options[:record_info], options[:record_title]))
+
+        # Append tag and policy breadcrumb if they exist
+        breadcrumbs.push(special_page_breadcrumb(@tagitems || @politems || @ownershipitems || @retireitems))
+
+        # Append title breadcrumb if they exist and not same as previous breadcrumb (eg "Editing name")
+        if @title && not_show_page? && @title != breadcrumbs.compact.last.try(:[], :title)
+          breadcrumbs.push(:title => @title)
+        end
+      else
+        # Append breadcrumb from title of the accordion (eg "Policies")
+        breadcrumbs.push(:title => accord_name, :key => accord_container, :action => "accordion_select") if features?
+
+        # Append breadcrumbs created from the tree (eg "All policies > Red Hat policies > Policy 1")
+        breadcrumbs.concat(build_breadcrumbs_from_tree)
+
+        if options[:include_record] && options[:record_info].present?
+          # Append breadcrumb created from record_info if included
+          breadcrumbs.push(:title => options[:record_info][options[:record_title]])
+        else
+          # Append breadcrumb from the title of right cell
+          breadcrumbs.push(special_page_breadcrumb(@tagitems || @politems || @ownershipitems || @retireitems))
+          breadcrumbs.push(:title => @right_cell_text) if @sb["action"]
+        end
+      end
+      breadcrumbs.compact
+    end
+
+    def build_breadcrumbs_no_explorer(record_info, record_title)
+      if record_info[record_title]
+        notshow = restful? ? "" : "show"
```

Comment:
Looks like `gtl_url` from generic show mixin, maybe extend that to optionally prepend the controller?